azurerm/resource_arm_front_door.go

```diff
@@ -0,0 +1,1459 @@
+	cacheQueryParameterStripDirective := v["cache_query_parameter_strip_directive"].(string)
+	backendPoolName := v["backend_pool_name"].(string)
+
+	useDynamicCompression := frontdoor.DynamicCompressionEnabledDisabled
+
+	if cacheUseDynamicCompression {
+		useDynamicCompression = frontdoor.DynamicCompressionEnabledEnabled
+	}
+
+	cacheConfiguration := &frontdoor.CacheConfiguration{
+		QueryParameterStripDirective: frontdoor.Query(cacheQueryParameterStripDirective),
+		DynamicCompression:           useDynamicCompression,
+	}
+
+	backend := &frontdoor.SubResource{
+		ID: utils.String(helper.GetFrontDoorSubResourceId(subscriptionId, resourceGroup, frontDoorName, "BackendPools", backendPoolName)),
+	}
+
+	forwardingConfiguration := frontdoor.ForwardingConfiguration{
+		ForwardingProtocol: frontdoor.ForwardingProtocol(forwardingProtocol),
+		CacheConfiguration: cacheConfiguration,
+		BackendPool:        backend,
+		OdataType:          frontdoor.OdataTypeMicrosoftAzureFrontDoorModelsFrontdoorForwardingConfiguration,
+	}
+
+	if customForwardingPath != "" {
+		forwardingConfiguration.CustomForwardingPath = utils.String(customForwardingPath)
+	}
+
+	return forwardingConfiguration
+}
+
+func flattenArmFrontDoorBackendPools(input *[]frontdoor.BackendPool) []map[string]interface{} {
+	if input == nil {
+		return make([]map[string]interface{}, 0)
+	}
+
+	output := make([]map[string]interface{}, 0)
+
+	for _, v := range *input {
+		result := make(map[string]interface{})
+
+		if id := v.ID; id != nil {
+			result["id"] = *id
+		}
+
+		if name := v.Name; name != nil {
+			result["name"] = *name
+		}
+
+		if properties := v.BackendPoolProperties; properties != nil {
+			result["backend"] = flattenArmFrontDoorBackend(properties.Backends)
+			result["health_probe_name"] = flattenArmFrontDoorSubResource(properties.HealthProbeSettings, "HealthProbeSettings")
+			result["load_balancing_name"] = flattenArmFrontDoorSubResource(properties.LoadBalancingSettings, "LoadBalancingSettings")
+		}
+
+		output = append(output, result)
+	}
+
+	return output
+}
+
+func flattenArmFrontDoorBackendPoolsSettings(input *frontdoor.BackendPoolsSettings) bool {
+	if input == nil {
+		return true
+	}
+
+	result := false
+
+	if enforceCertificateNameCheck := input.EnforceCertificateNameCheck; enforceCertificateNameCheck != "" {
+		if enforceCertificateNameCheck == frontdoor.EnforceCertificateNameCheckEnabledStateEnabled {
+			result = true
+		}
+	}
+
+	return result
+}
+
+func flattenArmFrontDoorBackend(input *[]frontdoor.Backend) []interface{} {
+	if input == nil {
+		return make([]interface{}, 0)
+	}
+
+	output := make([]interface{}, 0)
+
+	for _, v := range *input {
+		result := make(map[string]interface{})
+
+		if address := v.Address; address != nil {
+			result["address"] = *address
+		}
+		if backendHostHeader := v.BackendHostHeader; backendHostHeader != nil {
+			result["host_header"] = *backendHostHeader
+		}
+
+		if v.EnabledState == frontdoor.Enabled {
```

Comment:
We should replace this with:
```suggestion
		result["enabled"] = v.EnabledState == frontdoor.Enabled
```

Reply:
Fixed.